### PR TITLE
CVE-2010-2108 and CVE-2013-0841

### PR DESCRIPTION
**Pull request by MattThyng**
Filled out files regarding CVE-2010-2108 and CVE-2013-0841

### CODE REVIEW

File: cves/CVE-2010-2108.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ curated_instructions: |
   integrity checks on this file to make sure you fill everything out properly.
   If you are a student, we cannot accept your work as finished unless curated is
   set to true.
-curated: false
+curated: False
```
Comment (andymeneely) [inline cves/CVE-2010-2108.yml]: MUST be true for this to be graded properly.

Comment (MattThyng) [reply cves/CVE-2010-2108.yml]: Oops, I have fixed it. Curated is not true

File: cves/CVE-2010-2108.yml [hunk 2]
```diff
@@ -195,6 +221,22 @@ mistakes:
     they have written there. Are they doing those? Does the fix look proper?
 
     Use those questions to inspire your answer. Don't feel obligated to answer
-    every one. Write a thoughtful entry here that those ing the software
+    every one. Write a thoughtful entry here that those in the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    I think a few mistakes went into the creation of the vulnerability. First
+    of all, I believe their may have been miscommunication within the team.
```
Comment (mesternefeld) [inline cves/CVE-2010-2108.yml]:
```suggestion
    of all, I believe there may have been miscommunication within the team. 
```

File: cves/CVE-2013-0841.yml [hunk 3]
```diff
@@ -95,10 +111,13 @@ discovered:
...
     If there is no evidence as to how this vulnerability was found, then you may
     leave the entries blank except for "answer". Write down where you looked in "answer".
-  answer: 
-  date: 
-  automated: 
-  google: 
+  answer: |
+    This vulnerability was discovered by a user and reported to the development team.
+    The vulnerability could be shown when a patch put a multiplier on the index. Then
+    if the user navigated to www.pandora.com, the browser crashed.
```
Comment (mesternefeld) [inline cves/CVE-2013-0841.yml]: Is this specific to www.pandora.com or could we generalize this more? If it is for pandora, that is very interesting.

File: cves/CVE-2010-2108.yml [hunk 4]
```diff
@@ -106,8 +124,9 @@ subsystem:
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: 
-  name: 
+  answer: |
+    This was in the chrome subsystem. This is because it is within the chrome directory.
+  name: chrome
```
Comment (johnbyrneRIT) [inline cves/CVE-2010-2108.yml]: Chrome seems a little broad? I could be totally off here, but is there not something like a tabs subsystem?

File: cves/CVE-2010-2108.yml [hunk 5]
```diff
@@ -131,12 +153,14 @@ major_events:
 
     The event doesn't need to be directly related to this vulnerability, rather,
     we want to capture what the development team was dealing with at the time.
-  answer: 
+  answer: |
+    It appears that at the time the vulnerability was created, the team was focused
+    on updating chrome for the new OS X operating system for mac computers.
```
Comment (johnbyrneRIT) [inline cves/CVE-2010-2108.yml]: So is the vulnerability Mac specific, or did the changes made for Mac affect other operating systems too?

File: cves/CVE-2013-0841.yml [hunk 6]
```diff
@@ -198,4 +229,12 @@ mistakes:
...
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    This vulnerability was introduced during the implementation phase of development.
+    I believe this vulnerability was the result of a coding mistake, but it could be
+    the result of a design mistake as well. Due to the relatively simple nature of
+    this vulnerability I believe that it could have simply been a mistake done by the
+    programmer. Not properly checking array indexing is something can be easily
+    forgotten. If input validation was not part of the design, then this is a design
+    mistake as well. In order to prevent this vulnerability going forward, it must be
```
Comment (johnbyrneRIT) [inline cves/CVE-2013-0841.yml]: I think the fact that this was reported by a user is also notable. Array index checking is something that unit tests can usually catch, so it makes me wonder if the testing part of the development process is another area for improvement.

File: cves/CVE-2013-0841.yml [hunk 7]
```diff
@@ -32,7 +32,15 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  In the content-blocking funtionality of Google Chrome, there is a
```
Comment (johnbyrneRIT) [inline cves/CVE-2013-0841.yml]: I'm not totally sure what the chrome specific content-blocking functionality is. Is that pop-up blocking? Certificate verification?

File: cves/CVE-2013-0841.yml [hunk 8]
```diff
@@ -77,9 +92,10 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    There is no evidence of automated unit tests for this vulnerability.
```
Comment (johnbyrneRIT) [inline cves/CVE-2013-0841.yml]: I would like to give 3 upvotes to this CVE. It's kind of surprising that something like index checking made it all the way to the product.

File: cves/CVE-2013-0841.yml (hunk 3 above)
Comment (MattThyng) [reply cves/CVE-2013-0841.yml]: I do not believe it is specific for www.pandora.com, but www.pandora.com was able to create the specific environment where this bug happens. I'll not that.

File: cves/CVE-2010-2108.yml (hunk 5 above)
Comment (MattThyng) [reply cves/CVE-2010-2108.yml]: The change was not specific to Mac, the VCC was only created when they were updating for OS X

File: cves/CVE-2013-0841.yml (hunk 6 above)
Comment (MattThyng) [reply cves/CVE-2013-0841.yml]: I'll note that

File: cves/CVE-2013-0841.yml (hunk 7 above)
Comment (MattThyng) [reply cves/CVE-2013-0841.yml]: Well, it is the content-blocking of plugins, pop-ups and more

Comment (MattThyng) [reply cves/CVE-2013-0841.yml]: I'll be more specific

File: cves/CVE-2013-0841.yml (hunk 8 above)
Comment (MattThyng) [reply cves/CVE-2013-0841.yml]: I'll add that

File: cves/CVE-2010-2108.yml (hunk 4 above)
Comment (MattThyng) [reply cves/CVE-2010-2108.yml]: I put chrome as the subsystem because of the file structure. I think "browser" may be a better fit

**Pull request merged**